### PR TITLE
Update JSpecify annotation fully-qualified names

### PR DESCRIPTION
**Pull request by jyoo980**
Updates the `HandlerUtil` with the `org.jspecify.annotations.NonNull` annotation [here](https://jspecify.dev/docs/api/org/jspecify/annotations/NonNull.html)

### CODE REVIEW

**Comment by msridhar** [comment]
@jyoo980 what a coincidence, I just opened https://github.com/projectlombok/lombok/pull/3613 which does something similar but updates the copyable annotations.  Do you want to just merge my commit into this PR and then I can close the other one?

**Comment by jyoo980** [comment]
@msridhar sure, sounds good. I'll try to get your patch merged into mine today.

File: AUTHORS [hunk 1]
```diff
@@ -26,6 +27,7 @@ Kevin Chirls <kchirls@users.noreply.github.com>
 Liu DongMiao <liudongmiao@gmail.com>
 Luan Nico <luannico27@gmail.com>
 Maarten Mulders <mthmulders@users.noreply.github.com>
+Manu Sridharan <manu@cs.ucr.edu>
```
Comment (jyoo980) [inline AUTHORS:30]: @msridhar Let me know if you'd like another email listed.

**Comment by jyoo980** [comment]
Merging this should close #3613

File: AUTHORS (hunk 1 above)
Comment (msridhar) [reply AUTHORS:30]: Could we possibly use `msridhar@gmail.com`?  Thanks a lot!

**Comment by msridhar** [comment]
Thanks for merging in the other change so quick @jyoo980!  We may also want to re-title this PR to capture the new scope

**Review by msridhar — APPROVED** [review]
LGTM

**Comment by jyoo980** [comment]
@Rawi01 is there anything else I can do to help get this PR merged?

**Comment by rspilker** [comment]
This is part of the latest [edge release](https://projectlombok.org/download-edge), all feedback is welcome.

**Pull request merged**